apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -158,20 +205,17 @@ router.post(
         req.body.prompt,
         prompts[prompts.length - 1].html || '',
         prompts[prompts.length - 1].python || '',
-        qid,
+        qidFull,
         res.locals.authn_user.user_id,
         res.locals.authz_data.has_course_permission_edit,
       );
 
       if (result.htmlResult) {
-        res.send(
-          GenerationResults(
-            result.htmlResult,
-            result.pythonResult,
-            result.jobSequenceId,
-            res.locals,
-          ),
-        );
+        res.set({
+          'HX-Redirect': `${res.locals.urlPrefix}/ai_generate_question?qid=${req.body.unsafe_qid}`,
```

Comment:
redirect back to self- perhaps there's a better way to trigger page refresh from htmx? we need this to be an htmx request if we want to continue leeching off of loading-states.

Reply:
This now redirects to another page.